File: eCommerce/eCommerce.Services/Database/Seat.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using eCommerce.Services.Enums;

namespace eCommerce.Services.Database
{
    public class Seat
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(5)]
        public string RowLabel { get; set; } = string.Empty;

        public int SeatNumber { get; set; }

        public SeatType SeatType { get; set; } = SeatType.Regular;

        /// <summary>
        /// When <see cref="SeatType"/> is Couple, the adjacent seat blocked by this loveseat.
        /// </summary>
        public int? PartnerSeatId { get; set; }

        [ForeignKey("PartnerSeatId")]
        public Seat? PartnerSeat { get; set; }

        public bool IsActive { get; set; } = true;

        // Hall relationship
        public int HallId { get; set; }

        [ForeignKey("HallId")]
        public Hall Hall { get; set; } = null!;

        // Navigation property for reservation seats referencing this seat
        public ICollection<ReservationSeat> ReservationSeats { get; set; } = new List<ReservationSeat>();
    }
}

File: eCommerce/UI/ecommerce_desktop/lib/models/seat.dart
import '../core/enums/api_enums.dart';

class Seat {
  final int? id;
  final int? hallId;
  final String? rowLabel;
  final int? seatNumber;
  final int? seatType;
  final int? partnerSeatId;
  final int? spotsOccupied;
  final bool? isActive;

  Seat({
    this.id,
    this.hallId,
    this.rowLabel,
    this.seatNumber,
    this.seatType,
    this.partnerSeatId,
    this.spotsOccupied,
    this.isActive,
  });

  factory Seat.fromJson(Map<String, dynamic> json) {
    return Seat(
      id: json['id'] as int?,
      hallId: json['hallId'] as int?,
      rowLabel: json['rowLabel'] as String?,
      seatNumber: json['seatNumber'] as int?,
      seatType: json['seatType'] as int?,
      partnerSeatId: json['partnerSeatId'] as int?,
      spotsOccupied: json['spotsOccupied'] as int?,
      isActive: json['isActive'] as bool?,
    );
  }

  bool get isCouple => seatType == SeatTypes.couple;

  int get normalizedType =>
      seatType == SeatTypes.couple ? SeatTypes.couple : SeatTypes.regular;
}
